File: jvm/java-blocking-lambda/src/main/java/dev/restate/sdk/examples/LambdaFactory.java
package dev.restate.sdk.examples;

import dev.restate.sdk.lambda.LambdaRestateServer;
import dev.restate.sdk.lambda.LambdaRestateServerFactory;

public class LambdaFactory implements LambdaRestateServerFactory {
  @Override
  public LambdaRestateServer create() {
    return LambdaRestateServer.builder().withService(new Greeter()).build();
  }
}

File: jvm/kotlin-lambda/src/main/kotlin/dev/restate/sdk/examples/LambdaFactory.kt
package dev.restate.sdk.examples

import dev.restate.sdk.lambda.LambdaRestateServer
import dev.restate.sdk.lambda.LambdaRestateServerFactory

class LambdaFactory : LambdaRestateServerFactory {
    override fun create(): LambdaRestateServer {
        return LambdaRestateServer.builder().withService(Greeter()).build()
    }
}
